modules/programs/senpai.nix

```diff
@@ -4,7 +4,23 @@ with lib;
 
 let
   cfg = config.programs.senpai;
-  cfgFmt = pkgs.formats.yaml { };
+  imports = [
+    (mkRenamedOptionModule [ "programs" "senpai" "addr" ] [
+      "programs"
+      "senpai"
+      "address"
+    ])
+    (mkRenamedOptionModule [ "programs" "senpai" "nick" ] [
+      "programs"
+      "senpai"
+      "nickname"
+    ])
+    (mkChangedOptionModule [ "programs" "senpai" "no-tls" ] [
+      "programs"
+      "senpai"
+      "tls"
+    ] (config: !config.programs.senpai.no-tls))
+  ];
```

Comment:
Sadly this is not going to work because of https://github.com/NixOS/nixpkgs/issues/96006. (Well, and also because this is defined in a `let in` but not used)

I think the best we can do for now is add assertions that the old options aren't used...

Sorry for leading you to a wrong track, this corner of the module system isn't quite ready yet.